functionslab.js

```diff
@@ -0,0 +1,184 @@
+console.log('======question 1=======');
```

Comment:
I see some good use of reduce functions and other array methods to solve questions. I also saw some instance of hard coding which is not a good practice. Avoid setting numbers manually but instead use properties like length and such to set up the parameter for your codes. Good job overall and keep it up!